ExchangeRateApp/Controllers/CurrencyPairViewController.swift

```diff
@@ -7,116 +7,72 @@
 
 import UIKit
 
-protocol PickCurrencyDelegate {
-    func pickCurrencyPair(code: String, fullName: String, currency: String)
+protocol PickCurrencyDelegate: AnyObject {
+    func pickedCurrency(_: CurrencyCodeData)
 }
 
-
 class CurrencyPairViewController: UIViewController, UITableViewDataSource, UITableViewDelegate {
     
-    var isTapped: Bool = false
```

Comment:
Use declarative names that make sense, otherwise it's confusing and unreadable (for you as well, hence bunch of dead code - unused variables)